src/manager.spec.js

```diff
@@ -1,173 +1,845 @@
-describe('AdsManager', function () {
-
-  var ads;
+describe('AdManager', function() {
+  var AdManager, adManager;
+  var MockGoogleTag = require('mock_google_tag');
   var adUnits = require('bulbs.ads.units');
-  var MockGoogleTag = require('mockGoogleTag');
-  var sandbox = sinon.sandbox.create();
 
-  beforeEach(function () {
+  beforeEach(function() {
+    AdManager = require('./manager');
     window.googletag = new MockGoogleTag();
 
-    ads = require('./manager.js');
-    ads.init();
+    window.TARGETING = {
+      dfp_site: 'onion',
+      dfp_pagetype: 'homepage'
+    };
+    adManager = new AdManager();
   });
 
-  afterEach(function () {
-    sandbox.restore();
-    var added = document.querySelectorAll('body > div');
+  describe('new AdManager', function() {
+    it('has no slots yet', function() {
+      expect(adManager.slots).to.eql({});
+    });
 
-    for (var i = 0, l = added.length; i < l; i ++) {
-      var v = added[i];
-      document.body.removeChild(v);
-    }
+    it('has a default ad id', function() {
+      expect(adManager.adId).to.equal(0);
+    });
+
+    it('merges global TARGETING with ad unit dfp site param', function() {
+      expect(adManager.targeting).to.eql({
+        dfp_site: 'onion',
+        dfp_pagetype: 'homepage'
+      });
+    });
+
+    it('is not initialized', function() {
+      expect(adManager.initialized).to.be.false;
+    });
+
+    it('binds context for all event handlers', function() {
+      TestHelper.stub(AdManager.prototype, 'bindContext');
+      adManager = new AdManager();
+      expect(adManager.bindContext.called).to.be.true;
+    });
+
+    context('base defaults', function() {
+      it('reloads on resize', function() {
+        expect(adManager.options.doReloadOnResize).to.be.true;
+      });
+
+      it('sets filter slots by viewport based on ad unit config', function() {
+        expect(adManager.options.filterSlotsByViewport).to.equal(adUnits.settings.hasOwnProperty('filterSlotsByViewport'));
```

Comment:
Why is there a has own property check? Couldn't we just compare these two values?
```js
expect(adManager.options.filterSlotsByViewport).to.equal(adUnits.settingsfilterSlotsByViewport);
```